api/envoy/extensions/dynamic_modules/v3/dynamic_modules.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+package envoy.extensions.dynamic_modules.v3;
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.dynamic_modules.v3";
+option java_outer_classname = "DynamicModulesProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/dynamic_modules/v3;dynamic_modulesv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#protodoc-title: Dynamic Modules common configuration]
+
+// Configuration of a dynamic module. A dynamic module is a shared object file that can be loaded via dlopen
+// by a various Envoy extension points. Currently, only HTTP filter (envoy.filters.http.dynamic_modules) is supported.
+//
+// How the module is loaded is determined by the extension point that uses it. For example, the HTTP filter
+// loads the module with dlopen on the first creation of the filter chain for the same shared object file.
```

Comment:
sorry let me clarify my understanding - are you saying that you would expect Envoy to load all SOs (under specified directory or wherever configured) at the startup time of Envoy? I think that works for almost all use cases except that this will eliminate the use case where someone wants to add SOs without restarting Envoy. Personally we won't need such runtime-reload of SOs, but it seems likely that some people will ask for it in the future, though that can be supported incrementally later.

Reply:
https://github.com/envoyproxy/envoy/pull/36448/commits/487cca7e10cc8d53d54aab14b111e0ae6e957b1b done - lmk what you think!

Reply:
I just meant that it should load it at configuration load time, not at request time.

Reply:
ah ok thank you for clarifying. Then my documentation comment was wrong and misleading. Let me revert some changes and fix accordingly

Reply:
clarified that it's config load time, not on the request path!